ground/src/main/java/com/google/android/ground/persistence/remote/firebase/schema/SurveyConverter.kt

```diff
@@ -16,48 +16,41 @@
 
 package com.google.android.ground.persistence.remote.firebase.schema
 
-import com.google.android.ground.model.Survey
-import com.google.android.ground.model.imagery.TileSource
+import com.google.android.ground.model.Survey as SurveyModel
 import com.google.android.ground.model.job.Job
 import com.google.android.ground.persistence.remote.DataStoreException
+import com.google.android.ground.persistence.remote.firebase.protobuf.parseFrom
 import com.google.android.ground.persistence.remote.firebase.schema.JobConverter.toJob
+import com.google.android.ground.proto.Survey as SurveyProto
 import com.google.firebase.firestore.DocumentSnapshot
-import kotlinx.collections.immutable.toPersistentList
 import kotlinx.collections.immutable.toPersistentMap
 
-/** Converts between Firestore documents and [Survey] instances. */
+/** Converts between Firestore documents and [SurveyModel] instances. */
 internal object SurveyConverter {
 
   @Throws(DataStoreException::class)
-  fun toSurvey(doc: DocumentSnapshot): Survey {
+  fun toSurvey(doc: DocumentSnapshot, jobs: List<Job> = listOf()): SurveyModel {
     if (!doc.exists()) throw DataStoreException("Missing survey")
 
-    val pd =
-      DataStoreException.checkNotNull(doc.toObject(SurveyDocument::class.java), "surveyDocument")
+    val surveyFromProto = SurveyProto::class.parseFrom(doc, 1)
+    val surveyFromDocument = doc.toObject(SurveyDocument::class.java)
```

Comment:
Shouldn't we remove this now that we are parsing the firebase doc as proto?

Reply:
If you think it's safe to remove, I can do it now.

Reply:
Done.